### PR TITLE
Add new PR template

### PR DESCRIPTION
**Pull request by nd-02110114**
This PR is split from #2294

I added new PR template and improved CONTRIBUTING.md, MOLNET_PR_TEMPLATE.md and coding.rst.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
+- [ ] Documentations (modification for documents)
+
+## Checklist
+
+- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
+  - [ ] Run `yapf -i <modified file>` and check no errors
```
Comment (ncfrey) [inline .github/PULL_REQUEST_TEMPLATE.md:25]: One small comment here - should we specify the required version of yapf (0.22.0) in this template and some other places? I know this trips up new contributors occasionally.

Comment (rbharath) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: +1!

File: README.md [hunk 2]
```diff
@@ -21,11 +21,8 @@ materials science, quantum chemistry, and biology.
   - [Docker](#docker)
   - [From source](#from-source)
 - [Getting Started](#getting-started)
-- [Contributing to DeepChem](/CONTRIBUTING.md)
-  - [Code Style Guidelines](/CONTRIBUTING.md#code-style-guidelines)
-  - [Documentation Style Guidelines](/CONTRIBUTING.md#documentation-style-guidelines)
-  - [Gitter](#gitter)
```
Comment (rbharath) [inline README.md:27]: Could we add a link to the Gitter back in? It looks like it was incorrectly indented into the contribution section

File: docs/source/development_guide/coding.rst [hunk 3]
```diff
@@ -18,12 +18,29 @@ checking it in.
...
 
-Yapf is run on every pull request to make sure the formatting is correct, so if
+YAPF is run on every pull request to make sure the formatting is correct, so if
 you forget to do this the continuous integration system will remind you.
-Because different versions of yapf can produce different results, it is
+Because different versions of YAPF can produce different results, it is
 essential to use the same version that is being run on CI.  At present, that
 is 0.22.  We periodically update it to newer versions.
 
+Linting
```
Comment (rbharath) [inline docs/source/development_guide/coding.rst:27]: This is great!

**Review by rbharath — APPROVED** [review]
This is great! I've made a minor comment below but marked this as approved. Feel free to merge in once my and @ncfrey's comments are addressed.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (nd-02110114) [reply .github/PULL_REQUEST_TEMPLATE.md]: I fixed!

File: README.md (hunk 2 above)
Comment (nd-02110114) [reply README.md:27]: I add a link to the Gitter back in appropriate section.

**Comment by nd-02110114** [comment]
This PR modify only documents, so I merge in.

**Pull request merged**